google-beta/resource_pubsub_subscription.go

```diff
@@ -1,228 +1,369 @@
+		return nil, err
+	}
 
-	d.Partial(true)
+	// Replace import id for the resource id
+	id, err := replaceVars(d, config, "projects/{{project}}/subscriptions/{{name}}")
+	if err != nil {
+		return nil, fmt.Errorf("Error constructing id: %s", err)
+	}
+	d.SetId(id)
 
-	if d.HasChange("push_config") {
-		_, err := config.clientPubsub.Projects.Subscriptions.ModifyPushConfig(d.Id(), &pubsub.ModifyPushConfigRequest{
-			PushConfig: expandPubsubSubscriptionPushConfig(d.Get("push_config").([]interface{})),
-		}).Do()
+	return []*schema.ResourceData{d}, nil
+}
 
-		if err != nil {
-			return fmt.Errorf("Error updating subscription %q: %s", d.Get("name"), err)
-		}
+func flattenPubsubSubscriptionName(v interface{}, d *schema.ResourceData) interface{} {
+	if v == nil {
+		return v
 	}
+	return NameFromSelfLinkStateFunc(v)
+}
 
-	d.Partial(false)
-
-	return nil
+func flattenPubsubSubscriptionTopic(v interface{}, d *schema.ResourceData) interface{} {
+	if v == nil {
+		return v
+	}
+	return ConvertSelfLinkToV1(v.(string))
 }
 
-func resourcePubsubSubscriptionDelete(d *schema.ResourceData, meta interface{}) error {
-	config := meta.(*Config)
+func flattenPubsubSubscriptionLabels(v interface{}, d *schema.ResourceData) interface{} {
+	return v
+}
 
-	name := d.Id()
-	call := config.clientPubsub.Projects.Subscriptions.Delete(name)
-	_, err := call.Do()
-	if err != nil {
-		return err
+func flattenPubsubSubscriptionPushConfig(v interface{}, d *schema.ResourceData) interface{} {
+	if v == nil {
+		return nil
 	}
+	original := v.(map[string]interface{})
+	if len(original) == 0 {
+		return nil
+	}
+	transformed := make(map[string]interface{})
+	transformed["push_endpoint"] =
+		flattenPubsubSubscriptionPushConfigPushEndpoint(original["pushEndpoint"], d)
+	transformed["attributes"] =
+		flattenPubsubSubscriptionPushConfigAttributes(original["attributes"], d)
+	return []interface{}{transformed}
+}
+func flattenPubsubSubscriptionPushConfigPushEndpoint(v interface{}, d *schema.ResourceData) interface{} {
+	return v
+}
 
-	return nil
+func flattenPubsubSubscriptionPushConfigAttributes(v interface{}, d *schema.ResourceData) interface{} {
+	return v
 }
 
-func resourcePubsubSubscriptionStateImporter(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
-	config := meta.(*Config)
+func flattenPubsubSubscriptionAckDeadlineSeconds(v interface{}, d *schema.ResourceData) interface{} {
+	// Handles the string fixed64 format
+	if strVal, ok := v.(string); ok {
+		if intVal, err := strconv.ParseInt(strVal, 10, 64); err == nil {
+			return intVal
+		} // let terraform core handle it if we can't convert the string to an int.
+	}
+	return v
+}
 
+func expandPubsubSubscriptionName(v interface{}, d *schema.ResourceData, config *Config) (interface{}, error) {
 	project, err := getProject(d, config)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
+	return getComputedSubscriptionName(project, d.Get("name").(string)), nil
```

Comment:
`getComputedSubscriptionName` is also used in `iam_pubsub_subscription.go` and `resource_pubsub_subscription_iam_test.go`, should I inline there as well?

Reply:
Ah, I didn't notice those usages; I'd say to put the method into a handwritten file and inline the contents here so that we don't touch too many existing files but aren't depending on some other file from this resource.